ykrt/src/compile/jitc_yk/codegen/x64/lsregalloc.rs

```diff
@@ -262,24 +256,36 @@ impl<'a> LSRegAlloc<'a> {
         self.stack.size()
     }
 
+    /// Is the instruction at [iidx] dead?
+    pub(crate) fn is_inst_dead(&self, iidx: InstIdx) -> bool {
```

Comment:
Does "dead" mean "was alive and now is not", or could it also mean "has not been live yet, but will be"?

I think the former, since iirc this replaces something with a name like "is_still_alive_after()"?

Reply:
Good question. It means "has DCE identified it as equivalent to a tombstone?" "Dead" was my quick attempt to capture that, but I can see that it might not be entirely clear. I'm open to suggestions!

Reply:
Does `has_inst_died()` capture it perhaps? The past tense is an attempt to anchor the event in time.

Reply:
It's not exactly time based: it really means "has DCE identified this as dead". That *could* even happen conceptually during trace creation. So "is" is probably right. I think "died" might be the contentious bit: it could be that we should call is `is_inst_tombstone` or similar?

Reply:
Changed to `is_inst_tombstone` in the `fixup!` commits below. [Note: I accidentally pushed another change which I didn't intend, so force pushed that away.]